pkg/registry/apis/iam/hooks.go

```diff
@@ -168,3 +170,162 @@ func (b *IdentityAccessManagementAPIBuilder) AfterResourcePermissionCreate(obj r
 		}
 	}(rp.DeepCopy()) // Pass a copy of the object
 }
+
+// convertRolePermissionsToTuples converts role permissions (action/scope) to v1 TupleKey format
+// using the shared zanzana.ConvertRolePermissionsToTuples utility and common.ToAuthzExtTupleKeys
+func convertRolePermissionsToTuples(roleUID string, permissions []iamv0.CoreRolespecPermission) ([]*v1.TupleKey, error) {
+	// Convert IAM permissions to zanzana.RolePermission format
+	rolePerms := make([]zanzana.RolePermission, 0, len(permissions))
+	for _, perm := range permissions {
+		// Split the scope to get kind, attribute, identifier
+		kind, _, identifier := accesscontrol.SplitScope(perm.Scope)
+		rolePerms = append(rolePerms, zanzana.RolePermission{
+			Action:     perm.Action,
+			Kind:       kind,
+			Identifier: identifier,
+		})
+	}
+
+	// Translate to Zanzana tuples
+	openfgaTuples, err := zanzana.ConvertRolePermissionsToTuples(roleUID, rolePerms)
+	if err != nil {
+		return nil, err
+	}
+
+	// Convert directly to v1 tuples using common utility
+	v1Tuples := common.ToAuthzExtTupleKeys(openfgaTuples)
+
+	return v1Tuples, nil
+}
+
+// AfterCoreRoleCreate is a post-create hook that writes the core role permissions to Zanzana (openFGA)
+func (b *IdentityAccessManagementAPIBuilder) AfterCoreRoleCreate(obj runtime.Object, _ *metav1.CreateOptions) {
+	if b.zClient == nil {
+		return
+	}
+
+	role, ok := obj.(*iamv0.CoreRole)
+	if !ok {
+		return
+	}
```

Comment:
Do we care about using `zTickets` here (same as permissions hooks) for limiting concurrency?

Reply:
ztickets have been added :smile: , I still prefer Zickets